src/data/calendar.ts

```diff
@@ -71,7 +71,15 @@ export const fetchCalendarEvents = async (
     );
   });
 
-  const results = await Promise.all(promises);
+  let results;
+  try {
+    results = await Promise.all(promises);
```

Comment:
Now if one of the promises fail, everything fails. We should use `allSettled` so we can still show the working ones.
Also we should show an error message for the failing ones.

```suggestion
    results = await Promise.allSettled(promises);
```

Reply:
`allSettled()` requires a newer JS profile or a shim.

Where do you envision the error message, e.g. if the calendar is shown in a calendar card in a dashboard where one of two configured calendars failed?

Reply:
https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/es.promise.all-settled.js

The error could be above the `ha-calendar` element and dismissible.

Reply:
Or just don't use `await Promise.all` :-) but await in the step below

Reply:
> Or just don't use `await Promise.all` :-) but await in the step below

Not sure what you mean by that TBH. If we were to use `allSettled` would we then add the linked module as a real dep or just copy that coding in with a reference to the repo?

Reply:
And we also need a way for the caller of `fetchCalendarEvents()` to know about the errors, so just returning the `Promise` might not be sufficient, but instead we also need e.g. a failure counter. I never really took some time to read about this whole async `Promise` topic, so not really my jam yet...